### DISCUSSION TITLE
Use of System.Linq.Async.Queryable for LINQ queries

### DISCUSSION THREAD

**Discussion by ah1508** [nhibernate/nhibernate-core#2577] [Q&A] [answered]
Dear all,

I am looking for a standard way to get async results from a LINQ queries. I know about Nhibernate extension methods but also tried `System.Linq.Async.Queryable` (https://www.nuget.org/packages/System.Linq.Async.Queryable/, https://github.com/dotnet/reactive)

Examples : 

```csharp
IAsyncEnumerable<Blog> blogs = (from b in session.Query<Blog>() select b).ToAsyncEnumerable();
ValueTask<List<Blog>> list = blogs.ToListAsync();
```

```csharp
bool b = await (from b in session.Query<User>() where u.Username.Equals("...") select b).ToAsyncEnumerable().AnyAsync();
```
It works but I just ran basic tests, any known issues ? 

Thanks !

**Comment by hazzik** [comment, marked as answer]
There is only one issue: this code does not make any async queries.
